src/org/labkey/cds/CDSController.java

```diff
@@ -191,12 +194,21 @@ public Object execute(Object o, BindException errors) throws Exception
             ApiSimpleResponse response = new ApiSimpleResponse();
             List<Item> items = new ArrayList<>();
 
-            List<RSSFeed> feeds = RSSService.get().getFeeds(getContainer(), getUser());
+            User user = getUser();
+
+            // In order to display blog posts on Dataspace's public page where the user is not required to log-in, we need to create a
+            // Limited user with Reader role otherwise no feeds will be returned by RSSService.get().getFeeds()
+            if (getUser().isGuest())
```

Comment:
I don't think we care if the user is a guest or not but rather if the user had read access to the folder:

```suggestion
            if (getContainer().hasPermission(user, ReadPermission.class))
```

Reply:
getContainer().hasPermission(user, ReadPermission.class) evaluates to false. Guest user doesn't have a read permission, hence the reason to create a LimitedUser with ReaderRole.

Reply:
sorry missed a `!` in front of that conditional it should have read more like:
```
if (!getContainer().hasPermission(user, ReadPermission.class))
```
Unless we are really trying to distinguish between guest and non-guest users, checking the access level is the preferred route.

Reply:
Ah, got it! That makes sense, thank you for clarifying. Now updated!